activerecord/test/cases/associations/has_and_belongs_to_many_associations_test.rb

```diff
@@ -780,6 +780,16 @@ def test_assign_ids_ignoring_blanks
     assert_equal [projects(:active_record), projects(:action_controller)].map(&:id).sort, developer.project_ids.sort
   end
 
+  def test_singular_ids_are_reloaded_after_collection_concat
+    project = Project.new(name: "Serious Project")
+    project.developer_ids
+
+    developer = Developer.create(name: "Alberto Almagro")
+    project.developers << developer
+
+    assert_includes project.developer_ids, developer.id
```

Comment:
This assertion would pass without 19c80718afea8cf8f56ee68fcfbe325d54c02906.

Reply:
@kamipo Yes sorry, it seems I committed a previous version that used `Project.new` instead of `Project.create`. This version does fail without 19c8071. Thanks for double checking 🙏 ❤️